compiler/rustc_trait_selection/src/traits/select/confirmation.rs

```diff
@@ -286,6 +290,53 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
         ImplSourceBuiltinData { nested: obligations }
     }
 
+    fn confirm_transmutability_candidate(
+        &mut self,
+        obligation: &TraitObligation<'tcx>,
+    ) -> Result<ImplSourceBuiltinData<PredicateObligation<'tcx>>, SelectionError<'tcx>> {
+        debug!(?obligation, "confirm_transmutability_candidate");
+
+        let predicate = obligation.predicate;
+
+        let type_at = |i| predicate.map_bound(|p| p.trait_ref.substs.type_at(i));
+        let bool_at = |i| {
+            predicate
+                .skip_binder()
+                .trait_ref
+                .substs
+                .const_at(i)
+                .try_eval_bool(self.tcx(), obligation.param_env)
+                .unwrap()
```

Comment:
I'm guessing you'll need to gracefully handle this in the future (people will start throwing const generics or associated consts at this field)

Reply:
Resolved by https://github.com/rust-lang/rust/pull/92268/commits/ff0c538703e8dca92fe31fcc5d447443e28bffcb